File: sdk/samples/WFSamples/Technologies/Activities/Policy/SimplePolicy/VB/SimplePolicy/SimplePolicyWorkflow.vb
'---------------------------------------------------------------------
'  This file is part of the Windows Workflow Foundation SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'---------------------------------------------------------------------

Imports System
Imports System.CodeDom
Imports System.ComponentModel
Imports System.Workflow.Activities
Imports System.Workflow.Activities.Rules

Public Class SimplePolicyWorkflow
    Inherits SequentialWorkflowActivity

    ' Set orderValue and customerTypeValue to change the discount calculated by the policy
    Private orderValue As Double = 600.0
    Private customerTypeValue As CustomerType = CustomerType.Residential
    Private discount As Double = 0

    Private Sub OnWorkflowCompleted(ByVal sender As System.Object, ByVal e As System.EventArgs)
        Console.WriteLine("Order value = {0:c}", orderValue)
        Console.WriteLine("Customer type = {0}", customerTypeValue)
        Console.WriteLine("Calculated discount = {0}%", discount)
    End Sub

    Public Enum CustomerType
        Empty = 0
        Residential = 1
        Business = 2
    End Enum
End Class

File: sdk/samples/WFSamples/Technologies/Activities/Policy/SimplePolicy/CS/SimplePolicy/SimplePolicyWorkflow.cs
//---------------------------------------------------------------------
//  This file is part of the Windows Workflow Foundation SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//---------------------------------------------------------------------

using System;
using System.Workflow.Activities;
using System.CodeDom;
using System.Workflow.Activities.Rules;

namespace Microsoft.Samples.Workflow.SimplePolicy
{
    public sealed partial class SimplePolicyWorkflow : SequentialWorkflowActivity
    {
        // Set orderValue and customerType to change the discount calculated by the policy
        private double orderValue = 600.00;
        private CustomerType customerType = CustomerType.Residential;
        private double discount = 0;

        public SimplePolicyWorkflow()
        {
            InitializeComponent();
        }
        
        private void WorkflowCompleted(object sender, EventArgs e)
        {
            Console.WriteLine("Order value = {0:c}", orderValue);
            Console.WriteLine("Customer type = {0}", customerType);
            Console.WriteLine("Calculated discount = {0}%", discount);
        }

        public enum CustomerType
        {
            Empty = 0,
            Residential = 1,
            Business = 2
        }
    }
}
